tools/esdlcmd/esdlcmd_manifest.cpp

```diff
@@ -0,0 +1,1311 @@
+                return true;
+            if (isAbsolutePath(filePath))
+                return false;
+
+            for (std::string include : includes)
+            {
+                StringBuffer tryPath(include.c_str());
+                addNonEmptyPathSepChar(tryPath);
+                tryPath.append(filePath);
+                foundFile.setown(createIFile(tryPath));
+                if (foundFile->isFile() == fileBool::foundYes)
+                    return true;
+            }
+
+            foundFile.clear();
+            return false;
+        }
+
+        int skipNextEndTag(IXmlPullParser* xpp)
+        {
+            int type;
+            do
+            {
+                type = xpp->next();
+            }
+            while(type != XmlPullParser::END_TAG && type != XmlPullParser::END_DOCUMENT);
+            EndTag etag;
+            xpp->readEndTag(etag);
+
+            return type;
+        }
+
+        // To correctly enclose CDATA sections, any occurrence of the CDATA_END
+        // string must be replaced with the ENCODED_CDATA_END string
+        void encodeCdataEnd(const char* in, StringBuffer& out)
+        {
+            const char *str = in;
+            int offset = 0;
+
+            while(1)
+            {
+                const char *cDataEnd = strstr(str, CDATA_END);
+
+                if (cDataEnd != 0)
+                {
+                    out.append(in, offset, unsigned(cDataEnd - (in + offset)));
+                    // encode the found ]]>
+                    out.append(ENCODED_CDATA_END);
+                    // move past the end of the found ]]>
+                    offset = unsigned(cDataEnd - in) + 3;
+                    str = cDataEnd + 1;
+                    continue;
+                }
+                else
+                {
+                    out.append(in + offset);
+                    break;
+                }
+            }
+        }
+
+        int processInclude(IXmlPullParser* xpp, StartTag& root, IncludeType includeType)
+        {
+            const char* filename = nullptr;
+            StringBuffer content;
+            Owned<IFile> file;
+
+            // Do not copy out the <em:Include ... /> node itself
+            filename = root.getValue("file");
+
+            if (nullptr == filename)
+                throw makeStringExceptionV(ESDL_MANIFEST_ERR, "File attribute missing from Include statement on line #%d column #%d", xpp->getLineNumber(), xpp->getColumnNumber());
+
+            if (!locateFile(filename, includeSearchPaths, file))
+                throw makeStringExceptionV(ESDL_MANIFEST_ERR, "Unable to locate include file %s on line #%d column #%d", filename, xpp->getLineNumber(), xpp->getColumnNumber());
+
+            if (IncludeType::script == includeType)
+            {
+                StringBuffer rawFile;
+                rawFile.loadFile(file);
+                // Trim to ensure correct indentation
+                rawFile.trim();
+                if (!rawFile.length())
```

Comment:
I can see your point but wouldn't it be more harmful to silently output a binding missing scripts because you forgot to fill in the empty files? I'd think in general you'd expect included files to have some content more often than not.

Reply:
I would expect empty files to be less common. I'm just not convinced that an empty file should result in automatic failure. A warning message might be sufficient.

Reply:
The ESDL helper class already returns throws if the file is empty, however, this exception I'm throwing would be after the ESDLHelper has processed the file which should result in a valid defn. So if it is empty at that point I think it is an indication of some problem worthy of an exception.